### PR TITLE
first draft for lichen plagiarism detector

### PR DESCRIPTION
**Pull request by tushargr**
Tasks(remaining) to be done:-
- [ ] Creating lichen directories when the course is created.( to avoid permission errors). Migration part remaining.
- [ ]  ~Cronjob to create prior_assignment.txt and to run plagiarism~
- [x]   Color code that is rendered in code boxes (Assumed no overlap among match types)
- [x] Toggle button to toggle between left and right user

### CODE REVIEW

**Comment by tushargr** [comment]
@bmcutler  
for the first task, I have added few lines setup_sample_courses.py.
It will create lichen, lichen/config and lichen/provided_code folder at the time of course creation.

The owner for these dir will hwcron and group will be <course>_tas_www

for example in development course:-
drwxr-s---  4 hwcron  development_tas_www 4096 Jun 25 07:20 lichen
drwxrws--- 2 hwcron development_tas_www 4096 Jun 25 07:20 config
drwxrws--- 2 hwcron development_tas_www 4096 Jun 25 07:20 provided_code

Is it fine?

**Comment by tushargr** [comment]
@bmcutler 
I didn't get the thing with migrations. Can u elaborate why are we using migrations?
And the migration for the above change (creating lichen dirs for existing courses), will it be migration/system folder?

**Comment by tushargr** [comment]
@bmcutler 
add colours for the codebox , but I have assumed that any of start -end range for 3 types of match wont overlap for a user matches.json

File: migration/migrations/course/20180627121504_lichen_directories.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    course_dir          = os.path.join("/var/local/submitty/courses",semester,course)
+    lichen_dir          = os.path.join(course_dir,"lichen")
+    lichen_config_dir   = os.path.join(lichen_dir,"config")
+    lichen_provided_dir = os.path.join(lichen_dir,"provided_code")
+
+    # create the directories
+    os.system("mkdir -p "+lichen_dir)
+    os.system("mkdir -p "+lichen_config_dir)
+    os.system("mkdir -p "+lichen_provided_dir)
```
Comment (MasterOdin) [inline migration/migrations/course/20180627121504_lichen_directories.py]:
```
os.makedirs(lichen_dir, exist_ok=True)
os.makedirs(lichen_config_dir, exist_ok=True)
os.makedirs(lichen_provided_dir, exist_ok=True)
```

File: site/app/libraries/FileUtils.php [hunk 2]
```diff
@@ -395,8 +395,56 @@ public static function getContentType($filename){
...
+                $line = fgets($file);
+                if(trim(trim($line," "),"\n") == "") {
+                    break;
+                }
+                array_push($gradeables, trim(trim($line," "),"\n"));
+            }
+            $return[$sem][$course] = $gradeables;    
+            
+          }
+          fclose($file);
```
Comment (MasterOdin) [inline site/app/libraries/FileUtils.php]: What does the contents of this file look like?

Comment (tushargr) [reply site/app/libraries/FileUtils.php]: Contents will be like this:
/var/local/submitty/courses/f17/csci1100/submissions/:
hw01
hw02
hw03
hw04

/var/local/submitty/courses/f17/csci1200/submissions/:
final_exam
final_exam_notes
hw01
hw02

Comment (tushargr) [reply site/app/libraries/FileUtils.php]: The file is created using following command-
ls -1 /var/local/submitty/courses/*/*/submissions/ > /var/local/submitty/courses/gradeables_from_prior_terms.txt

This command will be later run by cronjob

**Comment by bmcutler** [comment]
Waiting on #2287

**Pull request merged**